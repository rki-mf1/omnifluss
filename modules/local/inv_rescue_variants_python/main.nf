process INV_RESCUE_VARIANTS_PYTHON {
    tag "$meta.id"

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.12' :
        'biocontainers/python:3.12' }"

    input:
    tuple val(meta), path(vcf)

    output:
    tuple val(meta), path('*.special_case_variant_mask.bed'), emit: bed
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    python $projectDir/bin/rescue_variants.py $vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rescue_variants.py: \$(python $projectDir/bin/rescue_variants.py --version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch "${prefix}.special_case_variant_mask.bed"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rescue_variants.py: \$(python $projectDir/bin/rescue_variants.py --version)
    END_VERSIONS
    """
}
