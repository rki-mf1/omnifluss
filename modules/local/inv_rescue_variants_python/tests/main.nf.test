nextflow_process {

    name "Test Process INV_RESCUE_VARIANTS_PYTHON"
    script "../main.nf"
    process "INV_RESCUE_VARIANTS_PYTHON"

    tag "modules"
    tag "modules_local"
    tag "inv_rescue_variants_python"

    test("Influenza - vcf - special case") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test.spec_case.filtered' ],
                    [ file(params.local_testdata_base_path + 'INV_illumina/vcf/test.spec_case.filtered.vcf.gz', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.bed).match("Snapshot-BED") },
                { assert snapshot(process.out.versions).match("Snapshot-VERSIONS") },
                { assert path(process.out.bed[0][1]).readLines().any{ it.contains("var1|B_/_H0N0|HA\t3\t4") } }
            )
        }
    }


    test("Influenza - vcf - special case - stub") {

        tag "stub"
        options '-stub'

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test.spec_case.filtered' ],
                    [ file(params.local_testdata_base_path + 'INV_illumina/vcf/test.spec_case.filtered.vcf.gz', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.bed).match("Snapshot-BED-stub") },
                { assert snapshot(process.out.versions).match("Snapshot-VERSIONS-stub") },
                { assert path(process.out.bed[0][1]).readLines().size() == 0 }
            )
        }
    }


    test("SARS-CoV-2 - vcf - no special case") {

        when {
            process {
                """
                input[0] =  Channel.of([
                                [ id:'test2' ],
                                [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz', checkIfExists: true) ]
                            ])
                """
            }
        }

        // the nf-core test data should not contain such a special case
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.bed).match() },
                { assert path(process.out.bed[0][1]).readLines().size() == 0 }
            )
        }
    }

}
