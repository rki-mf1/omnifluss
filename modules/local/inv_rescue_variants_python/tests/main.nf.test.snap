{
    "Snapshot-BED-stub": {
        "content": [
            [
                [
                    {
                        "id": "test.spec_case.filtered.vcf"
                    },
                    "test.spec_case.filtered.vcf.special_case_variant_mask.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-11T15:23:08.582335374"
    },
    "Snapshot-VERSIONS-stub": {
        "content": [
            [
                "versions.yml:md5,542aebb26490259942157e100eba7c8b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2025-02-13T16:35:17.094699981"
    },
    "Snapshot-BED": {
        "content": [
            [
                [
                    {
                        "id": "test.spec_case.filtered.vcf"
                    },
                    "test.special_case_variant_mask.bed:md5,bdd89429997e007414772c250cd704de"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-11T15:23:02.487412468"
    },
    "Snapshot-VERSIONS": {
        "content": [
            [
                "versions.yml:md5,542aebb26490259942157e100eba7c8b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2025-02-13T16:35:12.139841764"
    },
    "SARS-CoV-2 - vcf - no special case": {
        "content": [
            [
                [
                    {
                        "id": "test2"
                    },
                    "test2.special_case_variant_mask.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-11T15:36:13.594846046"
    }
}